### PR TITLE
oc: enhance deploymentconfig description

### PR DESCRIPTION
**Pull request by kargakis**
@smarterclayton @ironcladlou how about 
```
$ oc describe dc/router
Name:		router
Created:	8 minutes ago
Labels:		router=router
Annotations:	<none>
Latest Version:	1
Triggers:	Config
Strategy:	Rolling
Template:
  Selector:	router=router
  Replicas:	1
  Containers:
  router:
    Image:	openshift/origin-haproxy-router:v1.1.2
    QoS Tier:
      memory:	BestEffort
      cpu:	BestEffort
    Liveness Probe:
      HTTP request:
      Path: /healthz
      Port: 1936
      Host: localhost
      Scheme: HTTP
    Initial Delay: 10s
    Timeout: 1s
    Period: 10s
    SuccessThreshold: 1
    FailureThreshold: 3
    Readiness Probe:
      HTTP request:
      Path: /healthz
      Port: 1936
      Host: localhost
      Scheme: HTTP
    Initial Delay: 10s
    Timeout: 1s
    Period: 10s
    SuccessThreshold: 1
    FailureThreshold: 3
    Environment Variables:
      DEFAULT_CERTIFICATE:	-----BEGIN CERTIFICATE-----
MIIDEjCCAfqgAwIBAgIBDTANBgkqhkiG9w0BAQsFADAmMSQwIgYDVQQDDBtvcGVu
c2hpZnQtc2lnbmVyQDE0NTU2NDk1MTcwHhcNMTYwMjE2MTkwNTM0WhcNMTgwMjE1
MTkwNTM1WjAdMRswGQYDVQQDExIqLjEyNy4wLjAuMS54aXAuaW8wggEiMA0GCSqG
...

      OPENSHIFT_CERT_DATA:			
      OPENSHIFT_INSECURE:			false
      OPENSHIFT_KEY_DATA:			
      OPENSHIFT_MASTER:				https://localhost:8443
      ROUTER_EXTERNAL_HOST_HOSTNAME:		
      ROUTER_EXTERNAL_HOST_HTTPS_VSERVER:	
      ROUTER_EXTERNAL_HOST_HTTP_VSERVER:	
      ROUTER_EXTERNAL_HOST_INSECURE:		false
      ROUTER_EXTERNAL_HOST_PARTITION_PATH:	
      ROUTER_EXTERNAL_HOST_PASSWORD:		
      ROUTER_EXTERNAL_HOST_PRIVKEY:		/etc/secret-volume/router.pem
      ROUTER_EXTERNAL_HOST_USERNAME:		
      ROUTER_SERVICE_NAME:			router
      ROUTER_SERVICE_NAMESPACE:			default
      STATS_PASSWORD:				TGxmIeyfxk
      STATS_PORT:				1936
      STATS_USERNAME:				admin
Deployment #1 (latest):
	Name:		router-1
	Created:	8 minutes ago
	Status:		Complete
	Replicas:	1 current / 1 desired
	Selector:	deployment=router-1,deploymentconfig=router,router=router
	Labels:		openshift.io/deployment-config.name=router,router=router
	Pods Status:	1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Events:
  FirstSeen	LastSeen	Count	From				SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  8m		8m		1	{deploymentconfig-controller }		Normal		DeploymentCreated	Created new deployment "router-1" for version 1

```
cc: @jwforres @fabianofranz 

Fixes https://github.com/openshift/origin/issues/7330

### CODE REVIEW

**Comment by ironcladlou** [comment]
Looks like the tabbing for env var values is out of whack

**Comment by kargakis** [comment]
Printing certificates right is a noble goal to pursue

**Comment by ironcladlou** [comment]
I'm seeing misalignment on all the lines, not just the certificate

**Comment by kargakis** [comment]
Ah, sweet markdown. Those are printed fine in the terminal.

File: pkg/cmd/cli/describe/deployments.go [hunk 1]
```diff
@@ -260,16 +261,98 @@ func printDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec, w io.Writer)
...
-			formatLabels(convertEnv(container.Env)))
-	}
+	fmt.Fprintf(w, "  Containers:\n")
+	describeContainers(spec.Template.Spec, w)
+
 	return nil
 }
 
+// TODO: We should reuse this from upstream
+func describeContainers(spec kapi.PodSpec, w io.Writer) {
```
Comment (smarterclayton) [inline pkg/cmd/cli/describe/deployments.go]: Can we do that now? :)

File: pkg/cmd/cli/describe/deployments.go [hunk 2]
```diff
@@ -260,16 +261,98 @@ func printDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec, w io.Writer)
...
+
+		fmt.Fprintf(w, "    Environment Variables:\n")
+		for _, e := range container.Env {
+			fmt.Fprintf(w, "      %s:\t%s\n", e.Name, e.Value)
+		}
+	}
+}
+
+// TODO: needs to live upstream
+func describeProbe(probe kapi.Probe, w io.Writer) {
```
Comment (smarterclayton) [inline pkg/cmd/cli/describe/deployments.go]: I implemented probe differently, see my upstream PR

File: pkg/cmd/cli/describe/deployments.go (hunk 1 above)
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: The upstream describer needs a pod and returns info about its status. I don't think we can reuse it if we don't refactor it first.

File: pkg/cmd/cli/describe/deployments.go (hunk 2 above)
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: See my comments there

**Comment by kargakis** [comment]
```
$ oc describe dc
Name:		docker-registry
Created:	5 seconds ago
Labels:		docker-registry=default
Annotations:	<none>
Latest Version:	1
Triggers:	Config
Strategy:	Rolling
Template:
  Selector:	docker-registry=default
  Replicas:	1
  Containers:
  registry:
    Image:	openshift/origin-docker-registry:latest
    QoS Tier:
      cpu:	BestEffort
      memory:	BestEffort
    Liveness:	http-get http://:5000/healthz delay=10s timeout=5s period=10s #success=1 #failure=3
    Readiness:	http-get http://:5000/healthz delay=0s timeout=5s period=10s #success=1 #failure=3
    Environment Variables:
      OPENSHIFT_CA_DATA:	-----BEGIN CERTIFICATE-----
MIIC6jCCAdKgAwIBAgIBATANBgkqhkiG9w0BAQsFADAmMSQwIgYDVQQDDBtvcGVu
c2hpZnQtc2lnbmVyQDE0NTU4OTg5MzIwHhcNMTYwMjE5MTYyMjExWhcNMjEwMjE3
MTYyMjEyWjAmMSQwIgYDVQQDDBtvcGVuc2hpZnQtc2lnbmVyQDE0NTU4OTg5MzIw
ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDAZCapGzRMhdnYX94Ym8R5
cSj2kE5PUvhiP6I0jko19em8obZmn+/aUBVY76Z+cjL+BXBX3U4iSqh3Ah5UWBmJ
v7soxn4Qnt8O/FAW5ZqsNhAWQ8zeQOlcfv6vDulTTZ+R+cJcGadlE4X7CyQzJoNL
BAJ8MKLGsgcmXN7pr0274sP9dQMtr2mNJ0CR/y5LplZEKBiBN7WrB8fpwpeDE2TO
f3JOM84IIB7alRNKUjulB/b1flGPkfq5QRYVAgo8/bTys28Rzu+7sYlb03tRcr2T
...

      OPENSHIFT_MASTER:		https://10.0.2.15:8443
      REGISTRY_HTTP_ADDR:	:5000
      REGISTRY_HTTP_NET:	tcp
      REGISTRY_HTTP_SECRET:	9d5rOYm0nBpR79ajoNgu+g/rILU7xOYYK/wIJTICprI=
Deployment #1 (latest):
	Name:		docker-registry-1
	Created:	5 seconds ago
	Status:		Pending
	Replicas:	0 current / 0 desired
	Selector:	deployment=docker-registry-1,deploymentconfig=docker-registry,docker-registry=default
	Labels:		docker-registry=default,openshift.io/deployment-config.name=docker-registry
	Pods Status:	0 Running / 0 Waiting / 0 Succeeded / 0 Failed
Events:
  FirstSeen	LastSeen	Count	From				SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  5s		5s		1	{deploymentconfig-controller }			Normal		DeploymentCreated	Created new deployment "docker-registry-1" for version 1

```
@smarterclayton @ironcladlou

**Comment by kargakis** [comment]
@smarterclayton picked your changes from upstream, we will need the follow-up in order to use both helpers though.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**